### PR TITLE
Configure some components arguments from the pillar

### PR DESCRIPTION
**Pull request by mssola**
Signed-off-by: Miquel Sabaté Solà <msabate@suse.com>

### CODE REVIEW

**Comment by mssola** [comment]
The default for this flag is 5m, which is unreasonable imho.

**Comment by inercia** [comment]
I would prefer to leave the default values unless you have a strong reason...

**Comment by mssola** [comment]
@inercia waiting 5 minutes to re-schedule a pod from a node it is known by kubernetes to be down is unreasonable I think. I agree that 30 seconds is quite of a magic number. Maybe another way to solve this is to provide variables in the params.sls file for extra arguments for the controller-manager, kubelet, etc. I'm thinking on something like this (in params.sls):

```sls
kubernetes:
  args:
    kubelet: ""
    controller-manager: ""
    ...
```

What do you think ?

**Comment by inercia** [comment]
> What do you think ?

I think it is a good idea to allow users to customize the arguments for those components. However, I would stick with the default set of arguments unless strong reasons exist.

**Comment by inercia** [comment]
In your example, `kubernetes` is probably not a good root for the hierarchy, as everything here is related to kubernetes...

**Comment by mssola** [comment]
> I think it is a good idea to allow users to customize the arguments for those components. However, I would stick with the default set of arguments unless strong reasons exist.

Right, those parameters would be left empty by default, meaning that no extra arguments would be provided.

> In your example, kubernetes is probably not a good root for the hierarchy, as everything here is related to kubernetes...

True 😅

**Comment by inercia** [comment]
>> I think it is a good idea to allow users to customize the arguments for those components. However, I would stick with the default set of arguments unless strong reasons exist.

> Right, those parameters would be left empty by default, meaning that no extra arguments would be provided.

As we are into it, I would also add a variable for extra arguments for the Docker daemon (configured [here](https://github.com/kubic-project/salt/blob/master/salt/docker/init.sls#L18)).

**Comment by mssola** [comment]
@inercia updated. I'll squash commits when it's good to be merged.

File: salt/docker/init.sls [hunk 1]
```diff
@@ -15,16 +15,15 @@ docker:
     - require:
       - pkg: docker
 
-{% set docker_opt = "--iptables=false" %}
 {% set docker_reg = pillar.get('docker_registry_mirror', '') %}
```
Comment (inercia) [inline salt/docker/init.sls]: As you are already into this, maybe you could also put `docker_registry_mirror` under `docker` and have 

```
docker:
    # Extra arguments to be passed to the Docker daemon.
    args: '--iptables=false'
    registry_mirror: ''
```

Comment (mssola) [inline salt/docker/init.sls]: Sure thing.

**Comment by mssola** [comment]
@inercia done

**Pull request merged**